app/modules/code_provider/github/github_service.py

```
@@ -455,667 +455,667 @@ def _parse_link_header(self, link_header: str) -> Dict[str, str]:
+
+                        first_page_data = await response.json()
+                        all_installations.extend(first_page_data)
+                        break
+                except (ClientConnectorError, asyncio.TimeoutError) as net_err:
+                    attempt += 1
+                    if attempt >= max_attempts:
+                        logger.error(f"Network error contacting GitHub installations API: {net_err}")
+                        raise HTTPException(
+                            status_code=503,
+                            detail="Unable to reach GitHub API (installations). Please check network/proxy settings and try again.",
+                        )
```

Ruff (0.15.10)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)